luajit21.py

```diff
@@ -4023,3 +4023,27 @@ def invoke (self, args, from_tty):
 
 ltb()
 
+class ldumpstack(gdb.Command):
+    """This command takes a lua_State pointer and dumps all contents from
+it's stack.
+Usage: ldumpstack (lua_State *)"""
+
+    def __init__ (self):
+        super (ldumpstack, self).__init__("ldumpstack", gdb.COMMAND_USER)
+
+    def invoke (self, args, from_tty):
+        argv = gdb.string_to_argv(args)
+
+        if len(argv) != 1:
+            raise gdb.GdbError("usage: ltb <lua_State *>")
+
+        top = int(gdb.execute("printf \"%%d\", lua_gettop(%s)" % argv[0],
```

Comment:
No, you should not call userspace C functions directly. It won't work in core dumps, for example, and also unsafe for a live process since it might have unexpected side-effects or just won't be allowed in the current arbitrary process execution context. I'm getting the following error in my gdb when inspecting a core dump file:

```
(gdb) dump-all-timers
now is 1500918960355

timer node key=1500918962153, is_lua_timer=1, in -1798 msec
coroutine=0x41a52138. stack contents:
Traceback (most recent call last):
  File "luajit21.py", line 4041, in invoke
    to_string=True))
gdb.error: You can't do that without a process to debug.
Error occurred in Python command: You can't do that without a process to debug.
```